### DISCUSSION TITLE
Patch document in a browser and save the result

### DISCUSSION THREAD

**Discussion by RomanBorodatov** [dolanmiu/docx#2139] [Q&A] [answered]
Hi guys. First of all, thanks for this amazing library. Truly marvelous work!
I am going to use it for generating some reports in my React app. And even though I can create documents from scratch, I thought it would be nice to have some templates for these reports. 
While creating POC, I stumbled upon an issue with the`patchDocument` function. I can upload a local template and patch it, but can't save the file (from the browser). `patchDocument` returns Node Buffer and I am not sure how to handle it.
I am getting a feeling that adding polyfill with not solve the issue.

So, my question is:
Is it possible to patch the template in the browser and save it? If yes - please give me a hint. If not - do you plan to implement it?

Thanks

**Comment by dolanmiu** [comment, marked as answer]
I see, yes, I will make an update which will allow other formats other than node buffer to be exported

**Reply by dolanmiu** [reply]
This is now supported, patchDocument returns `Uint8Array` which is supported by browsers and node

**Reply by seeprogramming** [reply]
Hi @dolanmiu , Thank you in advance this is a great work. I am working on something and I am trying to use patchDocument in ReactJS , is there any snippet available where I can add images in existing template  using React JS OR is there any plan to add reactjs support for the same ?

and if any new version is coming for React Support , when can we expect it ?

**Reply by dolanmiu** [reply]
Will be available in v9 yes
